cli/src/alluxio.org/cli/cmd/exec/test_hms.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package exec
+
+import (
+	"alluxio.org/cli/env"
+	"alluxio.org/log"
+	"bytes"
+	"github.com/palantir/stacktrace"
+	"github.com/spf13/cobra"
+)
+
+var TestHms = &TestHmsCommand{
+	BaseJavaCommand: &env.BaseJavaCommand{
+		CommandName:   "testHms",
+		JavaClassName: "alluxio.cli.HmsTests",
+	},
+}
+
+type TestHmsCommand struct {
+	*env.BaseJavaCommand
+	metastore     string
+	database      string
+	tables        string
+	socketTimeout string
+}
+
+func (c *TestHmsCommand) Base() *env.BaseJavaCommand {
+	return c.BaseJavaCommand
+}
+
+func (c *TestHmsCommand) ToCommand() *cobra.Command {
+	cmd := c.Base().InitRunJavaClassCmd(&cobra.Command{
+		Use:   "testHms -m <arg> [-d <arg>] [-h] [-s <arg>] [-t <arg>]",
+		Short: "Test the configuration, connectivity, and permission of an existing hive metastore.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.Run(args)
+		},
+	})
+	cmd.Flags().StringVarP(&c.metastore, "metastore", "m", "",
+		"[required] Uri(s) to connect to hive metastore.")
+	cmd.Flags().StringVarP(&c.database, "database", "d", "",
```

Comment:
i noticed that all the flags do not have a default value set. right now the java code does more logic and may check "if user gives empty value, i will set a default value". we should shift all that logic to the go code and reduce the amount of validation in the java code. as a first step, we can update the go code without touching the java code; it is okay to have the check twice. it will make it easier to delete the logic from the java code in the future.

@Kai-Zhang can you help double check and review that we are reducing the dependency on the java code logic?

Reply:
Some commands specified default value with -h flag, while others didn't. Maybe need to use `-XX:+PrintCommandLineFlags` to check to default flag values in the java class.

Reply:
oh i have to read the java code to determine if there is a default value to set. i don't trust the `-h` flag because it prints a string output that may not actually be equal to the code logic. not sure about what the `-XX:+ PrintCommandLineFlags` flag will do, never seen it before